src/view.c

```diff
@@ -1624,6 +1624,56 @@ view_move_to_edge(struct view *view, enum view_edge direction, bool snap_to_wind
 	view_move(view, destination_x, destination_y);
 }
 
+struct output *
+view_find_output(struct view *view, const char *to)
+{
+	assert(view);
+	struct output *new_output;
+	if (!strcasecmp(to, "left")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_LEFT);
+	} else if (!strcasecmp(to, "right")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_RIGHT);
+	} else if (!strcasecmp(to, "up")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_UP);
+	} else if (!strcasecmp(to, "down")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_DOWN);
+	} else {
+		new_output = output_from_name(view->server, to);
+	}
+	return new_output;
+}
+
+void
+view_move_to_output(struct view *view, struct output *output)
+{
+	assert(view);
+	if (view->fullscreen) {
+		return;
+	}
+
+	view_invalidate_last_layout_geometry(view);
+	view_set_output(view, output);
+	if (view_is_floating(view)) {
+		struct wlr_box box = output_usable_area_in_layout_coords(output);
+		if (rc.placement_policy == LAB_PLACE_AUTOMATIC) {
+			if (placement_find_best(view, &box)) {
+				view_move(view, box.x, box.y);
+			} else {
+				view_center(view, &box);
+			}
+		} else {
+			view_center(view, &box);
+		}
```

Comment:
`placement_find_best` needs a box that describes the (expected) `width` and `height` of the window, and it overwrites `x` and `y` with the optimum location, so you need to give it a copy of the `pending` size rather than the full extent of the output.

As for `view_center`, it will center on the entire output when the second argument is `NULL`, so there is no need to figure anything more to provide it.

It might be nice to add an argument to `view_place_initiail` that tells it to skip the under-cursor placement step, so that both the existing `AutoPlace` action and this function can just call, *e.g.*, `view_place_initial(view, /* allow_cursor */ false);`. Otherwise:

```suggestion
		if (rc.placement_policy == LAB_PLACE_AUTOMATIC) {
			struct wlr_box &box = view->pending;
			if (placement_find_best(view, &box)) {
				view_move(view, box.x, box.y);
				return;
			} 
		} else {
			view_center(view, NULL);
		}
```

Reply:
Thanks for the clarification, that helped a lot. I think I got it now, at least from testing the behavior looked good. 
I tried to re-use  `view_place_initial` but unfortunately  that didn't worked together with resize.